src/daft-session/src/python.rs

```diff
@@ -93,11 +102,21 @@ impl PySession {
     pub fn set_namespace(&self, ident: Option<&PyIdentifier>) -> PyResult<()> {
         Ok(self.0.set_namespace(ident.map(|i| i.as_ref()))?)
     }
-}
 
-impl From<&PySession> for Session {
-    fn from(sess: &PySession) -> Self {
-        sess.0.clone()
+    #[pyo3(signature = (func, alias = None))]
+    pub fn attach_function(&self, func: PyObject, alias: Option<String>) -> PyResult<()> {
```

Comment:
signature matters slightly less here, but might as well make it match the public side